app/post.go

```diff
@@ -19,6 +20,13 @@ import (
 	"github.com/mattermost/mattermost-server/utils"
 )
 
+const (
+	PENDING_POST_IDS_CACHE_SIZE = 25000
+	PENDING_POST_IDS_CACHE_SEC  = 30
+)
+
+var seenPendingPostIds = utils.NewLruWithParams(PENDING_POST_IDS_CACHE_SIZE, "seenPendingPostIds", PENDING_POST_IDS_CACHE_SEC, "")
```

Comment:
The naming difficulty here is that `pendingPosts` suggests they are still pending, when in fact they will have completed up to 30 seconds ago.

Agree that this should probably be on the app server instead of a global.